### PR TITLE
UX: Remove count from more section links

### PR DESCRIPTION
**Pull request by tgxworld**
Less is more...

### CODE REVIEW

**Review by martin-brennan — APPROVED** [review]
This looks good, more or less :P

File: config/locales/client.en.yml [hunk 1]
```diff
@@ -4077,7 +4077,7 @@ en:
         one: "%{count} new"
         other: "%{count} new"
       toggle_section: "toggle section"
-      more_count: "%{count} more..."
+      more: "more..."
```
Comment (mcwumbly) [inline config/locales/client.en.yml:4080]: nit: I think we should capitalize this `More...`

Comment (tgxworld) [reply config/locales/client.en.yml]: Done :)

**Comment by tgxworld** [comment]
Thank you @martin-brennan

**Pull request merged**